frontend/catalyst/pennylane_extensions.py

```diff
@@ -151,20 +156,24 @@ def __init__(self, fn, *, method, h, argnum):
         self.h = h
         self.argnum = argnum
         if self.method != "fd":
-            assert isinstance(
-                self.fn, qml.QNode
-            ), "Only finite difference can compute higher order derivatives."
+            if not isinstance(self.fn, qml.QNode):
+                raise ValueError("Only finite difference can compute higher order derivatives.")
 
     def __call__(self, *args, **kwargs):
         """Specifies that an actual call to the differentiated function.
         Args:
             args: the arguments to the differentiated function
         """
         jaxpr = jax.make_jaxpr(self.fn)(*args)
-        assert len(jaxpr.eqns) == 1, "Grad is not well defined"
-        assert (
-            jaxpr.eqns[0].primitive == jprim.func_p
-        ), "Attempting to differentiate something other than a function"
+        if len(jaxpr.eqns) != 1:
+            raise TypeError(
+                f"Grad is not well defined for non-single Jax equations (got {len(jaxpr.eqns)})"
```

Comment:
I just encountered this issue actually and a good description is that grad is only supported on QNodes and other grads:
```suggestion
                f"catalyst.grad can only be used on QNodes and other grad calls"
```

I'm looking to change this in an upcoming PR to allow grad to be used on any function though :)